aspnetcore/mvc/controllers/application-model.md

```diff
@@ -12,7 +12,184 @@ ms.technology: aspnet
+
+Applying this to an action within the previous example's controller demonstrates how it overrides the controller-level convention:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/DescriptionAttributesController.cs?name=DescriptionAttributesController&highlight=9)]
+
+### Sample: Modifying the ParameterModel
+
+The following convention can be applied to action parameters to modify their `BindingInfo`. In this case, the convention specifies that the `BinderModelName` is "Special". This could be used to customize [model binding](https://docs.microsoft.com/aspnet/core/mvc/models/model-binding) behavior.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/SpecialParameterAttribute.cs)]
+
+The attribute is applied to any action parameter:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/ParameterModelController.cs?name=ParameterModelController&highlight=3)]
+
+### Sample: Modifying the ActionModel Name
+
+The following convention modifies the `ActionModel` to update the *name* of the action to which it is applied. The new name is provided as a parameter to the attribute. This new name is used by routing, so it will affect the route used to reach this action method.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/CustomActionNameAttribute.cs)]
+
+This attribute is applied to an action method in the `HomeController`:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/HomeController.cs?name=ActionModelConvention&highlight=2)]
+
+Even though the method name is `SomeName`, the attribute overrides the MVC convention of using the method name and replaces the action name with `MyCoolAction`. Thus, the route used to reach this action is `/Home/MyCoolAction`.
+
+### Sample: Custom Routing Convention
+
+You can use an `IApplicationModelConvention` to customize how routing works. For example, the following convention will incorporate Controllers' namespaces into their routes, replacing `.` in the namespace with `/` in the route:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/NamespaceRoutingConvention.cs)]
+
+The convention is added as an option in Startup.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Startup.cs?name=ConfigureServices&highlight=6)]
+
+> [!TIP]
+> If you need to add a convention to MVC separately from where MVC is configured (for instance, as part of your own [middleware](https://docs.microsoft.com/aspnet/core/fundamentals/middleware)), you can access `MvcOptions` using `services.Configure<MvcOptions>(c => c.Conventions.Add(YOURCONVENTION));`
+
+This sample only applies to convention to routes that aren't using attribute routing, and for demo purposes it's further constrained to only apply to controllers with "Namespace" in the name. This controller demonstrates the behavior applied by this convention:
```

Comment:
The sample applies this convention to routes that are not using attribute routing where the controller has  "Namespace" in the name. The following controller demonstrates this convention: